morepath/request.py

```diff
@@ -191,12 +199,12 @@ class (if this app doesn't handle them) to another app.
         if app is SAME_APP:
             app = self.app
 
-        info = app._get_deferred_mounted_path(obj)
+        info, app = app._get_deferred_mounted_path(obj)
```

Comment:
Reviewing it again I actually think my original comment is wrong. Your new proposed change makes little sense to me -- we can't go deferring the app first and then if the result isn't SAME_APP use the request's app.

But in your original proposed change, you first determine app, then follow any defers for linking, and then use that app. I think I misread line 202, as the app actually is also an input into this whole process that information is not completely lost.

So I think your original pull request is correct and we should merge it. Unless I now forgot my original objection!